ydb/core/blobstorage/dsproxy/dsproxy_get_impl.cpp

```diff
@@ -391,4 +466,103 @@ void TGetImpl::OnVPutResult(TLogContext &logCtx, TEvBlobStorage::TEvVPutResult &
     Step(logCtx, outVGets, outVPuts, outGetResult);
 }
 
+bool TGetImpl::CheckDataInconsistency(const TBlobState &blobState, const TRope &data) {
+    TStackVec<TRope, TypicalPartsInBlob> partData(Info->Type.TotalPartCount());
+    const bool isBlock42 = (Info->Type.GetErasure() == TBlobStorageGroupType::Erasure4Plus2Block);
+    if (isBlock42) {
+        ErasureSplit((TErasureType::ECrcMode)blobState.Id.CrcMode(), Info->Type, data, partData);
+    }
+    for (const auto &item : blobState.PartMap) {
+        if (item.Data.IsEmpty()) {
+            continue;
+        }
+        const TRope &partToCheck = isBlock42 ? partData[item.PartIdRequested - 1] : data;
+        if (item.Data != partToCheck) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool TGetImpl::FindCorruptedPart42(const TBlobState &blobState, const TRope &data, ui32 &outPartIndex) {
+    const auto &parts = blobState.PartMap;
+    auto crcMode = static_cast<TErasureType::ECrcMode>(blobState.Id.CrcMode());
+    TStackVec<TRope, TypicalPartsInBlob> selectedParts(Info->Type.TotalPartCount());
+    TStackVec<TRope, TypicalPartsInBlob> restoredParts(Info->Type.TotalPartCount());
+    TRope restoredBlob;
+
+    for (ui32 excludedPart = 0; excludedPart < parts.size(); ++excludedPart) {
+        if (parts[excludedPart].Data.IsEmpty()) {
+            continue;
+        }
+        std::fill(selectedParts.begin(), selectedParts.end(), TRope());
+        std::fill(restoredParts.begin(), restoredParts.end(), TRope());
+        ui32 restoreMask = 0;
+        ui32 selectedCount = 0;
+        for (ui32 i = 0; i < parts.size(); ++i) {
+            if (i == excludedPart || parts[i].Data.IsEmpty()) {
+                continue;
+            }
+            ui32 partId = parts[i].PartIdRequested - 1;
+            if (selectedParts[partId].IsEmpty()) {
+                selectedParts[partId] = parts[i].Data;
+                restoreMask |= (1 << partId);
+                ++selectedCount;
+            }
+            if (selectedCount == 4) {
+                break;
+            }
+        }
+        if (selectedCount != 4) {
+            continue;
+        }
+        
+        restoredBlob.clear();
+        ErasureRestore(crcMode, Info->Type, data.size(), &restoredBlob, selectedParts, restoreMask);
+        ErasureSplit(crcMode, Info->Type, restoredBlob, restoredParts);
+        
+        ui32 mismatch = 0;
+        for (ui32 i = 0; i < parts.size(); ++i) {
+            if (i == excludedPart || parts[i].Data.IsEmpty()) {
+                continue;
+            }
+            ui32 partId = parts[i].PartIdRequested - 1;
+            if (parts[i].Data != restoredParts[partId]) {
+                ++mismatch;
+            }
+        }
+        if (mismatch == 0) {
```

Comment:
maybe? 
```suggestion
        if (mismatch > 0) {
```

as far as I can understand the function returns if mismatch was found or not